commands/build.go

```diff
@@ -211,6 +219,61 @@ func (o *buildOptions) toDisplayMode() (progressui.DisplayMode, error) {
 	return progress, nil
 }
 
+func buildMetricAttributes(b *builder.Builder, options *buildOptions) attribute.Set {
+	return attribute.NewSet(
+		attribute.String("driver.name", options.builder),
+		attribute.String("driver.type", b.Driver),
+		attribute.Stringer("build.options.hash", &buildOptionsHash{buildOptions: options}),
+	)
+}
+
+// buildOptionsHash computes a hash for the buildOptions when the String method is invoked.
+// This is done so we can delay the computation of the hash until needed by OTEL using
+// the fmt.Stringer interface.
+type buildOptionsHash struct {
+	*buildOptions
+	result     string
+	resultOnce sync.Once
+}
+
+func (o *buildOptionsHash) String() string {
+	o.resultOnce.Do(func() {
+		target := o.target
+		contextPath := o.contextPath
+		dockerfile := o.dockerfileName
+		vcs := getGitRemoteURL(o.contextPath)
+
+		h := sha256.New()
+		for _, s := range []string{vcs, target, contextPath, dockerfile} {
+			_, _ = io.WriteString(h, s)
+			h.Write([]byte{0})
+		}
+		o.result = hex.EncodeToString(h.Sum(nil))
+	})
+	return o.result
+}
+
+// getGitRemoteURL returns the git remote url attribute for a context path.
+func getGitRemoteURL(contextPath string) string {
+	// Check if this is a valid path to begin with before attempting
+	// to run git. It might be a non-filesystem URL.
+	if _, err := os.Stat(contextPath); err != nil {
```

Comment:
I think we should do the URL/- detection to match `LoadInputs` and unless I'm missing something, same validation would need to be called for the `vcs` codepath.

If you think this is outside of scope, then create a follow-up issue for it and assign to yourself.

Reply:
I took a look at `LoadInputs` and it checks if the path is a local directory before checking for a URL. It seems to use this same code for that same functionality so I just exposed it and used that function instead. I've also included an explicit check for `-` to avoid attempting to call stat on that path and to avoid performing this logic at all in those circumstances.

We'll only do git remote url detection when it's a local directory.

Reply:
@jsternberg Looks good